sites/public/src/ListingView.tsx

```diff
@@ -74,13 +72,145 @@ export const ListingView = (props: ListingProps) => {
+    ami: t("listings.unit.ami"),
   }
 
   let groupedUnits: Record<string, React.ReactNode>[] = null
-  if (listing.unitsSummary !== undefined && listing.unitsSummary.length > 0) {
-    groupedUnits = getSummariesTableFromUnitsSummary(listing.unitsSummary)
-  } else if (listing.unitsSummarized !== undefined) {
-    groupedUnits = getSummariesTableFromUnitSummary(listing.unitsSummarized.byUnitTypeAndRent)
+  let hmiHeaders: TableHeaders
+  let hmiData: Record<string, React.ReactNode>[] = null
+  if (listing.unitGroups !== undefined && listing.unitGroups.length > 0) {
+    // unit group summary
+    groupedUnits = listing.unitSummaries.unitGroupSummary.map((group) => {
+      let rentRange = null
+      let rentAsPercentIncomeRange = null
+      if (group.rentRange && group.rentRange.min === group.rentRange.max) {
+        rentRange = group.rentRange.min
+      } else if (group.rentRange) {
+        rentRange = `${group.rentRange.min} - ${group.rentRange.max}`
+      }
+
+      if (rentRange) {
+        rentRange = (
+          <span>
+            <strong>{rentRange}</strong> {t("t.perMonth")}
+          </span>
+        )
+      }
+
+      if (
+        group.rentAsPercentIncomeRange &&
+        group.rentAsPercentIncomeRange.min === group.rentAsPercentIncomeRange.max
+      ) {
+        rentAsPercentIncomeRange = group.rentAsPercentIncomeRange.min
+      } else if (rentAsPercentIncomeRange) {
+        rentAsPercentIncomeRange = `${group.rentAsPercentIncomeRange.min} - ${group.rentAsPercentIncomeRange.max}`
+      }
+
+      if (rentAsPercentIncomeRange) {
+        rentAsPercentIncomeRange = (
+          <span>
+            <strong>{rentAsPercentIncomeRange}%</strong> {t("t.income")}
+          </span>
+        )
+      }
+
+      let availability = null
+
+      if (group.unitVacancies > 0) {
+        availability = (
+          <div>
+            <strong>{group.unitVacancies} </strong>
+            {group.unitVacancies === 1 ? t("lisitngs.vacantUnit") : t("listings.vacantUnits")}
+            {" &"}
+          </div>
+        )
+      }
+
+      availability = (
+        <>
+          {availability}
+          <strong>
+            {group.openWaitlist ? t("listings.waitlist.open") : t("listings.waitlist.closed")}
+          </strong>
+        </>
+      )
+
+      let ami = null
+
+      if (
+        group.amiPercentageRange &&
+        group.amiPercentageRange.min === group.amiPercentageRange.max
+      ) {
+        ami = `${group.amiPercentageRange.min}%`
+      } else {
+        ami = `${group.amiPercentageRange.min} - ${group.amiPercentageRange.max}%`
+      }
+
+      return {
+        unitType: (
+          <>
+            {group.unitTypes
+              .map<React.ReactNode>((type) => (
+                <strong key={type}>{t(`listings.unitTypes.${type}`)}</strong>
+              ))
+              .reduce((acc, curr) => [acc, ", ", curr])}
+          </>
+        ),
+        rent:
+          rentRange || rentAsPercentIncomeRange ? (
```

Comment:
I find this a little hard to read, here is one suggestion:
let rentCol = ""
if (rentRange) rentCol = rentRange
if (rentAsPercentIncomeRange) rentCol = `${rentRange}${rentRange.length && ", "}${rentAsPercentIncomeRange}

Reply:
Updating to:
```
let rent: React.ReactNode = null

if (rentRange && rentAsPercentIncomeRange) {
  rent = (
    <>
      {rentRange}, {rentAsPercentIncomeRange}
    </>
  )
} else if (rentRange) {
  rent = rentRange
} else if (rentAsPercentIncomeRange) {
  rent = rentAsPercentIncomeRange
}
```